### PR TITLE
[CHIA-298] Optimize DL autoinserts.

### PR DESCRIPTION
**Pull request by fchirica**

### CODE REVIEW

**Comment by fchirica** [comment]
PR improves in CI from 1000 inserts in 36s to 25000 inserts in 52s. Locally on my laptop, I can do 100000 single batch inserts in 22s now.

File: chia/_tests/core/data_layer/test_data_store.py [hunk 1]
```diff
@@ -1567,6 +1598,40 @@ async def test_benchmark_batch_insert_speed(
         )
 
 
+@datacases(
+    BatchesInsertBenchmarkCase(
+        count=50,
+        batch_count=200,
+        limit=195,
+    ),
+)
```
Comment (altendky) [inline chia/_tests/core/data_layer/test_data_store.py:1607]: Were we intending to provide more parametrizations?

**Review by altendky — CHANGES_REQUESTED** [review]

File: chia/data_layer/data_store.py [hunk 2]
```diff
@@ -1313,6 +1313,20 @@ async def clean_node_table(self, writer: Optional[aiosqlite.Connection] = None)
...
+        while True:
+            assert len(queue) > 0
+            node = queue.pop(0)
+            if isinstance(node, InternalNode):
+                left_node = await self.get_node(node.left_hash)
+                right_node = await self.get_node(node.right_hash)
+                queue.append(left_node)
+                queue.append(right_node)
+            elif isinstance(node, TerminalNode):
+                return node
```
[the remark this answers is not in the document]
Comment (altendky) [reply chia/data_layer/data_store.py:1328]: There seems to be some more complexity in our query along with the desire to not only have breadth first but also to terminate early after finding the first leaf node.  If we think we can still manage something better, we'll take it on in a separate PR.

**Review by altendky — DISMISSED** [review]

**Review by altendky — APPROVED** [review]

**Review by altendky — CHANGES_REQUESTED** [review]
we are talking through back-compat

**Review by altendky — CHANGES_REQUESTED** [review]
suggested alternate form, feel free to close these conversations if you prefer to keep it as is

**Review by altendky — DISMISSED** [review]

**Review by emlowe — APPROVED** [review]

**Review by altendky — APPROVED** [review]

**Pull request merged**